jujugui/static/gui/src/app/components/search-results/search-results.js

```diff
@@ -21,100 +21,44 @@ with this program.  If not, see <http://www.gnu.org/licenses/>.
 YUI.add('search-results', function(Y) {
 
   juju.components.SearchResults = React.createClass({
-    search: function() {
-      /*
-       * XXX: replace this placeholder and its dummy data with live call to
-       * search service.
-       */
-      return {
-        hasResults: true,
-        solutionsCount: 5,
-        nameWidth: 'foobar',
+    getInitialState: function() {
+      // XXX: Implement "loading" state that differentiates between 0 search
+      // results and an empty results object.
+      return {data: {}}
+    },
+
+    componentDidMount: function() {
```

Comment:
I believe that this is only called the first time it's rendered, so subsequent renders will not perform any searches